source/common/upstream/cluster_manager_impl.cc

```diff
@@ -264,11 +264,13 @@ ClusterManagerStats ClusterManagerImpl::generateStats(Stats::Scope& scope) {
 }
 
 void ClusterManagerImpl::postInitializeCluster(Cluster& cluster) {
-  if (cluster.hosts().empty()) {
-    return;
+  for (size_t i = 0; i < cluster.prioritySet().hostSetsPerPriority().size(); ++i) {
```

Comment:
nit: Any reason to not just iterate directly through the returned vector here? Why do we need to get the size and then call get each time? (Same question if this happens elsewhere)

Reply:
Oh yeah, now that host set latches its own priority I can rewrite these.